heartbeat/process/messageProcessor.go

```diff
@@ -85,14 +87,26 @@ func (mp *MessageProcessor) verifySignature(hbRecv *data.Heartbeat) error {
 		return err
 	}
 
-	copiedHeartbeat := *hbRecv
-	copiedHeartbeat.Signature = nil
-	buffCopiedHeartbeat, err := mp.marshalizer.Marshal(&copiedHeartbeat)
-	if err != nil {
-		return err
+	pid, sig := mp.networkShardingCollector.GetPidAndSignatureFromPk(hbRecv.Pubkey)
+	if pid == nil || sig == nil {
```

Comment:
same thing here about what I have mentioned on consensus. I know I agree about this approach, but please add a TODO to have a single component for this (with reference to the consensus counter part), maybe including the aforementioned cached sig and pk bytes?

Reply:
Done.